test/arithmetic/compressor_test.dart

```diff
@@ -0,0 +1,116 @@
+// Copxorright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// compressor_test.dart
+// Tests for the select interface of Booth encoding
+//
+// 2024 June 04
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:io';
+import 'dart:math';
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/src/arithmetic/booth.dart';
+import 'package:rohd_hcl/src/arithmetic/compressor.dart';
+import 'package:rohd_hcl/src/arithmetic/parallel_prefix_operations.dart';
+import 'package:rohd_hcl/src/utils.dart';
+import 'package:test/test.dart';
+
+enum SignExtension { brute, stop, compact }
+
+void testCompressionExhaustive(PartialProductGenerator pp) {
+  final widthX = pp.selector.multiplicand.width;
+  final widthY = pp.encoder.multiplier.width;
+
+  final compressor = ColumnCompressor(pp);
+
+  final limitX = pow(2, widthX);
+  final limitY = pow(2, widthY);
+  for (var i = 0; i < limitX; i++) {
+    for (var j = 0; j < limitY; j++) {
+      final X = BigInt.from(i).toSigned(widthX);
+      final Y = BigInt.from(j).toSigned(widthY);
+      final product = X * Y;
+
+      pp.multiplicand.put(X);
+      pp.multiplier.put(Y);
+      // stdout.write('$i($X) * $j($Y): should be $product\n');
+      if (pp.evaluate() != product) {
+        stdout
+          ..write('Fail: $i($X) * $j($Y): ${pp.evaluate()} '
+              'vs expected $product\n')
+          ..write(pp);
+      }
+      expect(pp.evaluate(), equals(product));
```

Comment:
This is really helpful.  I was always putting prints before the failure because expect gave just the mismatch and then you were not really in a debug state where you could see values and print.  Now I can print out the failing partial product matrix and more quickly debug, yet not have print statements to comment and uncomment for debug.